ms2query/query_from_sqlite_database.py

```diff
@@ -135,61 +170,124 @@ def get_index_of_inchikeys(list_of_inchikeys: List[str],
+                    WHERE identifier_1 in ({identifier_string_1})
+                    and identifier_2 in ({identifier_string_2})
+                    UNION
+                    SELECT identifier_1, identifier_2, tanimoto_score
+                    FROM tanimoto_scores
+                    WHERE identifier_1 in ({identifier_string_2})
+                    and identifier_2 in ({identifier_string_1})
+                    ;
                     """
     cur = conn.cursor()
     cur.execute(sqlite_command)
-    result_list = []
-    for result in cur:
-        result_list.append(result)
-
-    # The data is changed to pd.DataFrame twice and then added together. So
-    # that both the tanimoto score is given for 1,2 and 2,1. Thereby
-    # duplicating the data, but making the lookup easier for other functions.
-    scores_normal_identifiers = pd.DataFrame(result_list,
-                                             columns=["identifier_1",
-                                                      "identifier_2",
-                                                      "tanimoto_score"])
-    scores_reversed_identifiers = pd.DataFrame(result_list,
-                                               columns=["identifier_2",
-                                                        "identifier_1",
-                                                        "tanimoto_score"])
-    result_dataframe_melt_structure = pd.concat([scores_normal_identifiers,
-                                                 scores_reversed_identifiers])
-
-    # Changes the structure of the database from a melt structure to a matrix
-    result_dataframe = pd.pivot_table(result_dataframe_melt_structure,
-                                      columns="identifier_1",
-                                      index="identifier_2",
-                                      values="tanimoto_score")
+    results = cur.fetchall()
+    result_dict = {}
+    for result in results:
+        identifier_1 = result[0]
+        identifier_2 = result[1]
+        tanimoto_score = result[2]
+        # Store the tanimoto score twice also with reversed combination
+        result_dict[(identifier_1, identifier_2)] = tanimoto_score
+        result_dict[(identifier_2, identifier_1)] = tanimoto_score
+    # Create matrix with tanimoto scores
+    tanimoto_scores = []
+    for identifier_1 in list_of_identifiers_1:
+        current_row = []
+        for identifier_2 in list_of_identifiers_2:
+            tanimoto_score = result_dict[(identifier_1, identifier_2)]
+            current_row.append(tanimoto_score)
+        tanimoto_scores.append(current_row)
+    # Store as pd.dataframe
+    result_dataframe = pd.DataFrame(data=tanimoto_scores,
+                                    index=list_of_identifiers_1,
+                                    columns=list_of_identifiers_2)
     conn.close()
     return result_dataframe
+
+
+def get_part_of_metadata_from_sqlite(sqlite_file_name: str,
```

Comment:
```suggestion
def get_metadata_from_sqlite(sqlite_file_name: str,
```

Reply:
I think that function could be named (and maybe also in its functions) a bit more generic.

Reply:
I agree, see next comment for how I implemented this.